src/analysis/convex/basic.lean

```diff
@@ -1380,66 +1637,81 @@ closure_operator.mk₃
 
 variable (s)
 
-lemma subset_convex_hull : s ⊆ convex_hull s :=
+lemma subset_convex_hull : s ⊆ convex_hull k s :=
 convex_hull.le_closure s
 
-lemma convex_convex_hull : convex (convex_hull s) :=
+lemma convex_convex_hull : convex k (convex_hull k s) :=
 closure_operator.closure_mem_mk₃ s
 
 variable {s}
 
-lemma convex_hull_min (hst : s ⊆ t) (ht : convex t) : convex_hull s ⊆ t :=
+lemma convex_hull_min (hst : s ⊆ t) (ht : convex k t) : convex_hull k s ⊆ t :=
 closure_operator.closure_le_mk₃_iff (show s ≤ t, from hst) ht
 
-lemma convex_hull_mono (hst : s ⊆ t) : convex_hull s ⊆ convex_hull t :=
+lemma convex_hull_mono (hst : s ⊆ t) : convex_hull k s ⊆ convex_hull k t :=
 convex_hull.monotone hst
 
-lemma convex.convex_hull_eq {s : set E} (hs : convex s) : convex_hull s = s :=
+lemma convex.convex_hull_eq {s : set E} (hs : convex k s) : convex_hull k s = s :=
 closure_operator.mem_mk₃_closed hs
 
 @[simp]
 lemma convex_hull_empty :
-  convex_hull (∅ : set E) = ∅ :=
+  convex_hull k (∅ : set E) = ∅ :=
 convex_empty.convex_hull_eq
 
 @[simp]
 lemma convex_hull_empty_iff :
-  convex_hull s = ∅ ↔ s = ∅ :=
+  convex_hull k s = ∅ ↔ s = ∅ :=
 begin
   split,
   { intro h,
     rw [←set.subset_empty_iff, ←h],
-    exact subset_convex_hull _ },
+    exact subset_convex_hull k _ },
   { rintro rfl,
     exact convex_hull_empty }
 end
 
 @[simp] lemma convex_hull_nonempty_iff :
-  (convex_hull s).nonempty ↔ s.nonempty :=
+  (convex_hull k s).nonempty ↔ s.nonempty :=
 begin
   rw [←ne_empty_iff_nonempty, ←ne_empty_iff_nonempty, ne.def, ne.def],
   exact not_congr convex_hull_empty_iff,
 end
 
 @[simp]
-lemma convex_hull_singleton {x : E} : convex_hull ({x} : set E) = {x} :=
+lemma convex_hull_singleton {x : E} : convex_hull k ({x} : set E) = {x} :=
 (convex_singleton x).convex_hull_eq
 
-lemma convex.convex_remove_iff_not_mem_convex_hull_remove {s : set E} (hs : convex s) (x : E) :
-  convex (s \ {x}) ↔ x ∉ convex_hull (s \ {x}) :=
+lemma convex.convex_remove_iff_not_mem_convex_hull_remove {s : set E} (hs : convex k s) (x : E) :
+  convex k (s \ {x}) ↔ x ∉ convex_hull k (s \ {x}) :=
 begin
   split,
   { rintro hsx hx,
     rw hsx.convex_hull_eq at hx,
     exact hx.2 (mem_singleton _) },
   rintro hx,
-  suffices h : s \ {x} = convex_hull (s \ {x}), { convert convex_convex_hull _ },
-  exact subset.antisymm (subset_convex_hull _) (λ y hy, ⟨convex_hull_min (diff_subset _ _) hs hy,
+  suffices h : s \ {x} = convex_hull k (s \ {x}), { convert convex_convex_hull k _ },
+  exact subset.antisymm (subset_convex_hull k _) (λ y hy, ⟨convex_hull_min (diff_subset _ _) hs hy,
     by { rintro (rfl : y = x), exact hx hy }⟩),
 end
 
-lemma affine_map.image_convex_hull (f : E →ᵃ[ℝ] F) :
-  f '' (convex_hull s) = convex_hull (f '' s) :=
+lemma is_linear_map.image_convex_hull {f : E → F} (hf : is_linear_map k f) :
```

Comment:
I think you are accidentally reintroducing old proofs and dropping new ones here.

If you look carefully at #9057 you'll see that I factored `is_linear_map.image_convex_hull` and `linear_map.image_convex_hull` through `affine_map.image_convex_hull` and I think we should not revert that change.

Reply:
This is actually on purpose. We must because affine stuff is currently less general than linear stuff. This will change once we generalize convexity to affine spaces.

Reply:
Is this because we require an underlying group structure for an affine model space, rather than a monoid?

If that's the case, then this single change might be a good initial PR to canvass opinion on where to draw the line in generalising convexity.

Reply:
I don't remember the details off the top of my head, but yeah pretty sure you're right.